File: python/0349.py
# Link: https://leetcode.com/problems/intersection-of-two-arrays/description/
# Runtime: 39 ms
# Memory: 16.8 MB


class Solution:
    def intersection(self, nums1: List[int], nums2: List[int]) -> List[int]:
        return list(set(nums1) & set(nums2))

File: c++/0349.cpp
// Link: https://leetcode.com/problems/intersection-of-two-arrays/description/
// Runtime: 4 ms
// Memory: 13.22 MB

class Solution {
public:
  std::vector<int> intersection(const std::vector<int> &nums1,
                                const std::vector<int> &nums2) {
    std::unordered_map<int, int> valueAndFrequency;
    for (const auto &val : nums1)
      valueAndFrequency[val]++;

    std::vector<int> intersectionArray;
    for (const auto &val : nums2) {
      if (valueAndFrequency.find(val) != valueAndFrequency.end()) {
        intersectionArray.push_back(val);
        valueAndFrequency.erase(val);
      }
    }

    return intersectionArray;
  }
};
